src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -631,3 +681,1111 @@ def delete_project(self, project_id: UUIDLike) -> GlobusHTTPResponse:
+            "domain_constraints_include": domain_constraints_include,
+            "domain_constraints_exclude": domain_constraints_exclude,
+        }
+
+        return self.post("/v2/api/policies", data={"policy": body})
+
+    def update_policy(
+        self,
+        policy_id: UUIDLike,
+        *,
+        project_id: UUIDLike | utils.MissingType = utils.MISSING,
+        authentication_assurance_timeout: int | utils.MissingType = utils.MISSING,
+        display_name: str | utils.MissingType = utils.MISSING,
+        description: str | utils.MissingType = utils.MISSING,
+        domain_constraints_include: (
+            t.Iterable[str] | None | utils.MissingType
+        ) = utils.MISSING,
+        domain_constraints_exclude: (
+            t.Iterable[str] | None | utils.MissingType
+        ) = utils.MISSING,
+    ) -> GlobusHTTPResponse:
+        """
+        Update a policy. Requires the ``manage_projects`` scope.
+
+        :param policy_id: ID of the policy to update
+        :type policy_id: str or uuid
+
+        :param project_id: ID of the project for the new policy
+        :type project_id: str or uuid
+        :param authentication_assurance_timeout: Number of seconds within which someone
+            must have authenticated to satisfy the policy
+        :type authentication_assurance_timeout: int
+        :param display_name: A user-friendly name for the policy
+        :type display_name: str
+        :param description: A user-friendly description to explain the purpose of the
+            policy
+        :type description: str
+        :param domain_constraints_include: A list of domains that can satisfy the policy
+        :type domain_constraints_include: iterable of str or None
+        :param domain_constraints_exclude: A list of domains that can not satisfy the
+            policy
+        :type domain_constraints_exclude: iterable of str or None
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> policy_id = ...
+                    >>> r = ac.update_policy(scope_id, display_name="Greenlight Policy")
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.update_policy
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/policies/{policy_id}``
+
+                .. extdoclink:: Update Policy
+                    :ref: auth/reference/#update_policy
+        """
+        body: dict[str, t.Any] = {
+            "authentication_assurance_timeout": authentication_assurance_timeout,
+            "display_name": display_name,
+            "description": description,
+            "domain_constraints_include": domain_constraints_include,
+            "domain_constraints_exclude": domain_constraints_exclude,
+        }
+
+        if not isinstance(project_id, utils.MissingType):
+            body["project_id"] = str(project_id)
```

Comment:
I see what's going on here, with `UUID | str | MissingType` -- and of course you need to `str()` it if it's a `UUID`.

:+1: to this as a way of handling it for now.
I need to think more about what we can do. Maybe the SDK should just support stringifying `UUID`s when we prepare a payload and strip out the `MISSING` values. It's a really common type for us... Your input is welcome, but no change needed here!

Reply:
FYI, we _just_ merged a change that makes SDK auto-`str()` any UUIDs.
I think these cases can change now to
```python
body["project_id"] = project_id
```

IMO any updates to this style are completely optional though. The current code works either way, which is the main thing.

---

As for `is MISSING`, I wish there were a nicer way to handle that for mypy. (`Literal` is wonky with custom types and doesn't work...)
I still haven't figured out what yet. Maybe we'll take a leaf out of sqlalchemy's book for naming things which look similar to language keywords, like this?
```python
if MISSING.is_(foo): ...
```
But I'm far from decided yet.